src/client_side.cc

```diff
@@ -1342,9 +1387,11 @@ parseHttpRequest(ConnStateData *csd, const Http1::RequestParserPointer &hp)
      *  - remote interception with PROXY protocol
      *  - remote reverse-proxy with PROXY protocol
      */
-    if (csd->transparent()) {
+    if (csd->switchedToHttps()) {
+        http->uri = csd->prepareTlsSwitchingURL(hp);
+    } else if (csd->transparent() || csd->switchedToHttps()) {
```

Comment:
This switchedToHttps() condition can never be true, if it was the if-statement above would have run instead.
```suggestion
    } else if (csd->transparent()) {
```

Reply:
addressed at 19f00737